language/move-binary-format/src/compatibility.rs

```diff
@@ -7,7 +7,16 @@ use crate::{
     file_format_common::VERSION_5,
     normalized::Module,
 };
-use std::collections::BTreeSet;
+use std::{
+    collections::BTreeSet,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+/// Whether to treat friends as private during compatibility check.
+static TREAT_FRIEND_AS_PRIVATE: AtomicBool = AtomicBool::new(false);
```

Comment:
I understand we are in a rush, but controlling this using an (effectively) mutable global variable really seems like an anti-pattern to me. Can we just pass this in as a parameter?

Notice that we already have an interface that passes in a verifier config
```
pub fn new_with_verifier_config(
        natives: impl IntoIterator<Item = (AccountAddress, Identifier, Identifier, NativeFunction)>,
        verifier_config: VerifierConfig,
    ) -> VMResult<Self> {
        Ok(Self {
            runtime: VMRuntime::new(natives, verifier_config)
                .map_err(|err| err.finish(Location::Undefined))?,
        })
    }
```
We can extend this easily by defining a VM config and change `new_with_verifier_config` to `new_with_vm_config`
```
struct MoveVMConfig {
    verifier_config: VerifierConfig,
    treat_friend_as_private: bool,
}
```